### PR TITLE
#151 Parent-Student Base Profile and API Prefs for #23

### PR DESCRIPTION
**Pull request by Pooja-Codes**
- Parent Profile View added with UI and the Preferences Section of Profile is integrated with the API.
- Updated Navbar for Profile.
- Added mitt package for event handling. @adelaidemz

### CODE REVIEW

**Review by Pooja-Codes — COMMENTED** [review]
I checked from my side! LGTM!

**Comment by simaoalicechen** [comment]
Please connect this with an issue.

**Comment by simaoalicechen** [comment]
Expect a little delay from my side on this pr's review. Other people can review before I start.

**Review by Pooja-Codes — COMMENTED** [review]
Verified from my side! LGTM!

File: src/frontend/shortlist/package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
     "@popperjs/core": "^2.11.6",
     "axios": "^1.1.3",
     "bootstrap": "^5.2.2",
+    "mitt": "^3.0.0",
```
Comment (simaoalicechen) [inline src/frontend/shortlist/package.json:18]: mitt is a smart move. Be careful about the Daniel's way. If no conflict, I am good with this.

Comment (simaoalicechen) [reply src/frontend/shortlist/package.json:18]: @adelaidemz Do you want to add a comment in the file?

File: src/frontend/shortlist/src/components/layout/profile/UserPreference.vue [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    },
+  },
+  mounted() {
+    axios
+      .post(
+        "https://api.shortlist.nyc/account/metadata",
+        {},
+        {
+          headers: {
+            "x-shortlist-account": "e44278fec4984522b156cae1a8b3981d",
```
Comment (simaoalicechen) [inline src/frontend/shortlist/src/components/layout/profile/UserPreference.vue:27]: This is hard coded. It is a good for now. Could you please add a comment for later to add api call?

File: src/frontend/shortlist/src/components/layout/profile/UserPreference.vue [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      prefupdateflag: false,
+    };
+  },
+  methods: {
+    updateflag() {
+      this.prefupdateflag = !this.prefupdateflag;
+    },
+  },
+  mounted() {
+    axios
```
Comment (yalexublue) [inline src/frontend/shortlist/src/components/layout/profile/UserPreference.vue:21]: emit up to view then post to keep current file fomat

**Review by simaoalicechen — APPROVED** [review]
LGTM!

**Review by yalexublue — APPROVED** [review]
LGTM

File: src/frontend/shortlist/package.json (hunk 1 above)
Comment (Pooja-Codes) [reply src/frontend/shortlist/package.json:18]: The updates are adaptable to accept future changes by Daniel.

File: src/frontend/shortlist/src/components/layout/profile/UserPreference.vue (hunk 2 above)
Comment (Pooja-Codes) [reply src/frontend/shortlist/src/components/layout/profile/UserPreference.vue:27]: Currently, We are using one user later we will structure for more/every by type.

File: src/frontend/shortlist/src/components/layout/profile/UserPreference.vue (hunk 3 above)
Comment (Pooja-Codes) [reply src/frontend/shortlist/src/components/layout/profile/UserPreference.vue:21]: Nice note for flow understanding to communicate with API.

**Pull request merged**